src/main/java/org/jabref/preferences/AiPreferences.java

```diff
@@ -13,203 +14,350 @@
+    private final StringProperty mistralAiApiBaseUrl;
+    private final StringProperty huggingFaceApiBaseUrl;
+
+    private final ObjectProperty<EmbeddingModel> embeddingModel;
+    private final StringProperty instruction;
     private final DoubleProperty temperature;
-    private final IntegerProperty messageWindowSize;
+    private final IntegerProperty contextWindowSize;
     private final IntegerProperty documentSplitterChunkSize;
     private final IntegerProperty documentSplitterOverlapSize;
     private final IntegerProperty ragMaxResultsCount;
     private final DoubleProperty ragMinScore;
 
-    public AiPreferences(boolean enableChatWithFiles, String openAiToken, ChatModel chatModel, EmbeddingModel embeddingModel, boolean customizeSettings, String systemMessage, double temperature, int messageWindowSize, int documentSplitterChunkSize, int documentSplitterOverlapSize, int ragMaxResultsCount, double ragMinScore) {
-        this.enableChatWithFiles = new SimpleBooleanProperty(enableChatWithFiles);
-        this.openAiToken = new SimpleStringProperty(openAiToken);
-
-        this.customizeSettings = new SimpleBooleanProperty(customizeSettings);
+    public AiPreferences(PreferencesService preferencesService,
+                         boolean enableAi,
+                         AiProvider aiProvider,
+                         String openAiChatModel,
+                         String mistralAiChatModel,
+                         String huggingFaceChatModel,
+                         String openAiApiToken,
+                         String mistralAiApiToken,
+                         String huggingFaceApiToken,
+                         boolean customizeExpertSettings,
+                         String openAiApiBaseUrl,
+                         String mistralAiApiBaseUrl,
+                         String huggingFaceApiBaseUrl,
+                         EmbeddingModel embeddingModel,
+                         String instruction,
+                         double temperature,
+                         int contextWindowSize,
+                         int documentSplitterChunkSize,
+                         int documentSplitterOverlapSize,
+                         int ragMaxResultsCount,
+                         double ragMinScore
+    ) {
+        this.preferencesService = preferencesService;
+
+        this.enableAi = new SimpleBooleanProperty(enableAi);
+
+        this.aiProvider = new SimpleObjectProperty<>(aiProvider);
+
+        this.openAiChatModel = new SimpleStringProperty(openAiChatModel);
+        this.mistralAiChatModel = new SimpleStringProperty(mistralAiChatModel);
+        this.huggingFaceChatModel = new SimpleStringProperty(huggingFaceChatModel);
+
+        this.openAiApiToken = new SimpleStringProperty(openAiApiToken);
+        this.mistralAiApiToken = new SimpleStringProperty(mistralAiApiToken);
+        this.huggingFaceApiToken = new SimpleStringProperty(huggingFaceApiToken);
```

Comment:
I think, it is now confusing that at the beginning `""` is stored.

Can you please factor out a class `AiApiKeys` class having the three keys.